nautobot/ipam/tests/test_models.py

```diff
@@ -59,6 +59,64 @@ def setUpTestData(cls):
             status=vmint_status,
         )
 
+    def test_removing_ip_addresses_containing_host_device_primary_ip_not_allowed(self):
+        """
+        Validate we cannot remove an IPAddress from an Interface that is the host Device's primary ip.
+        """
+        dev_ip_addr = IPAddress.objects.create(address="192.0.2.1/24")
```

Comment:
Should we use an arbitrary factory-generated IPAddress here and below instead of creating instances specifically for these tests?